src/MySqlConnector/Serialization/SerializationUtility.cs

```diff
@@ -47,5 +51,30 @@ public static int GetBytesPerCharacter(CharacterSet characterSet)
 				return 1;
 			}
 		}
+
+		public static void SetKeepalive(Socket socket, uint keepAliveTimeSeconds)
+		{
+			// Always use the OS Default Keepalive settings
+			socket.SetSocketOption(SocketOptionLevel.Socket, SocketOptionName.KeepAlive, true);
```

Comment:
I think that always enabling it to use the OS Default would be sensible.  It allows the OS to clean up old sockets should the connection go away.  If data is consistently being sent across the network, Keepalive packets will never need to be sent.

[Windows](https://blogs.technet.microsoft.com/nettracer/2010/06/03/things-that-you-may-want-to-know-about-tcp-keepalives/) and [Linux](http://www.tldp.org/HOWTO/html_single/TCP-Keepalive-HOWTO/#configuringkernel) both default to send the first Keepalive packet after 2 hours of inactivity on the socket.